compositor_render/src/transformations/builtin/transformations.rs

```diff
@@ -1,57 +1,88 @@
 use std::sync::Arc;
 
 use compositor_common::scene::{
-    builtin_transformations::{BuiltinTransformation, TransformToResolution},
-    ShaderParam,
+    builtin_transformations::{BuiltinTransformationSpec, TextureLayout, TransformToResolution},
+    Resolution, ShaderParam,
 };
 
 use crate::{
     renderer::{WgpuCtx, WgpuError},
     transformations::shader::Shader,
+    utils::rgba_to_wgpu_color,
 };
 
 pub struct BuiltinTransformations {
     transform_resolution: ConvertResolutionTransformations,
+    fixed_position_layout: FixedPositionLayout,
 }
 
 impl BuiltinTransformations {
     pub fn new(wgpu_ctx: &Arc<WgpuCtx>) -> Result<Self, WgpuError> {
         Ok(Self {
             transform_resolution: ConvertResolutionTransformations::new(wgpu_ctx)?,
+            fixed_position_layout: FixedPositionLayout::new(wgpu_ctx)?,
         })
     }
 
-    pub fn shader(&self, transformation: &BuiltinTransformation) -> Arc<Shader> {
+    pub fn shader(&self, transformation: &BuiltinTransformationSpec) -> Arc<Shader> {
         match transformation {
-            BuiltinTransformation::TransformToResolution(TransformToResolution::Stretch) => {
+            BuiltinTransformationSpec::TransformToResolution(TransformToResolution::Stretch) => {
                 self.transform_resolution.stretch.clone()
             }
-            BuiltinTransformation::TransformToResolution(TransformToResolution::Fill) => {
+            BuiltinTransformationSpec::TransformToResolution(TransformToResolution::Fill) => {
                 self.transform_resolution.fill.clone()
             }
-            BuiltinTransformation::TransformToResolution(TransformToResolution::Fit(_)) => {
+            BuiltinTransformationSpec::TransformToResolution(TransformToResolution::Fit(_)) => {
                 self.transform_resolution.fit.clone()
             }
+            BuiltinTransformationSpec::FixedPositionLayout { .. } => {
+                self.fixed_position_layout.0.clone()
+            }
         }
     }
 
-    pub fn params(transformation: &BuiltinTransformation) -> Option<ShaderParam> {
+    pub fn params(
+        transformation: &BuiltinTransformationSpec,
+        output_resolution: &Resolution,
+    ) -> Option<ShaderParam> {
         match transformation {
-            BuiltinTransformation::TransformToResolution(_) => None,
+            BuiltinTransformationSpec::TransformToResolution(_) => None,
+            BuiltinTransformationSpec::FixedPositionLayout {
+                textures_layouts: textures_specs,
+                ..
+            } => {
+                let mut layouts = Vec::new();
+
+                for TextureLayout {
+                    top,
+                    left,
+                    rotation,
+                } in textures_specs
+                {
+                    layouts.push(ShaderParam::I32(
```

Comment:
In wgsl you have list of structs, but here you have list of ints. It works because memory layout is the same, but this is not correct